cpp/dolfinx/fem/Form.h

```diff
@@ -422,46 +422,37 @@ class Form
       const iterator& tagged_facets_begin, const iterator& tagged_facets_end,
       const std::vector<int>& tags)
   {
-    // When a mesh is not ghosted by cell, it is not straightforward
-    // to distinguish between (i) exterior facets and (ii) interior
-    // facets that are on a partition boundary. If there are no
-    // ghost cells, build a set of owned facts that are ghosted on
-    // another process to help determine if a facet is on an
-    // exterior boundary.
-    int tdim = topology.dim();
-    assert(topology.index_map(tdim));
-    assert(topology.index_map(tdim - 1));
-    const std::vector<std::int32_t> fwd_shared_facets
-        = topology.index_map(tdim)->overlapped()
-              ? std::vector<std::int32_t>()
-              : topology.index_map(tdim - 1)->shared_indices();
+    const std::vector<std::int32_t> boundary_facets
+        = mesh::exterior_facet_indices(topology);
 
+    // Create list of tagged boundary facets
+    std::vector<std::int32_t> tagged_ext_facets;
+    std::set_intersection(tagged_facets_begin, tagged_facets_end,
```

Comment:
This requires that the input arrays are sorted. Is this guaranteed?

Any preconditions on the function input with regard to sorting should be documented.

Reply:
Yes, the input arrays are already sorted. I'll add documentation